lib/deep-mark-optional/index.ts

```diff
@@ -0,0 +1,29 @@
+import { HasParsablePath } from "../has-parsable-path";
+import { MarkOptional } from "../mark-optional";
+import { Paths } from "../paths";
+import { Prettify } from "../prettify";
+import { UnionToTuple } from "../union-to-tuple";
+
+type RecursiveDeepMarkOptionalSinglePath<Type, KeyPath> = Type extends object
+  ? KeyPath extends `${infer PropertyKey}.${infer RestKeyPath}`
+    ? {
+        [Key in keyof Type]: Key extends PropertyKey
+          ? Prettify<RecursiveDeepMarkOptionalSinglePath<Type[Key], RestKeyPath>>
+          : Type[Key];
+      }
+    : Prettify<MarkOptional<Type, KeyPath & keyof Type>>
+  : Type;
+
+type RecursiveDeepMarkOptional<Accumulator, KeyPaths extends Array<string>> = KeyPaths extends [
+  infer KeyPath,
+  ...infer RestKeyPaths,
+]
+  ? RestKeyPaths extends Array<string>
```

Comment:
Replace `Array<string>` with `string[]` to stay consistent with `DeepMarkRequired`.

```suggestion
type RecursiveDeepMarkOptional<Accumulator, KeyPaths extends string[]> = KeyPaths extends [
  infer KeyPath,
  ...infer RestKeyPaths,
]
  ? RestKeyPaths extends string[]
```